components/script/dom/htmlformelement.rs

```diff
@@ -746,10 +747,24 @@ impl HTMLFormElement {
         let enctype = submitter.enctype();
         let method = submitter.method();
 
-        // Step 17-21
-        let target_attribute_value = submitter.target();
+        // Step 17
+        let target_attribute_value =
+            if submitter.is_submit_button() && submitter.target() != DOMString::new() {
+                Some(submitter.target())
+            } else {
+                get_element_target(&(submitter.form_owner().unwrap()).upcast::<Element>())
```

Comment:
I believe this is the line that is causing us troubles. Let's use this instead:
```
let form = submitter.form_owner().as_ref().map(|form| &**form).unwrap_or(self);
get_element_target(form.upcast::<Element>())
```

Reply:
Does this go on the else condition?

Reply:
@jdm I get an error about a temporary value dropped while borrowed when I do this.

Reply:
@avr1254 Try:
```
let form_owner = submitter.form_owner();
let form = form_owner.as_ref().map(|form| &**form).unwrap_or(self);
get_element_target(form.upcast::<Element>())
```

Reply:
That did it. Nice!